src/model/card.ts

```diff
@@ -1,47 +1,177 @@
+     * </>
+     */
     protected domCreator() {
         const container = document.createElement('div');
         container.className = 'cards';
 
         const displayTopic = document.createElement('span');
+        container.className = 'sec-topic';
         displayTopic.textContent = this.topic;
-        
+
         const displayBody = document.createElement('span');
+        container.className = 'sec-body';
         displayBody.textContent = this.body;
 
+        const addTagButton = document.createElement('button');
+        addTagButton.textContent = 'AddTag';
+        addTagButton.className = "addtagbutton";
+        addTagButton.onclick = () => this.addTag();
+
         const removeButton = document.createElement('button');
         removeButton.textContent = 'Delete';
         removeButton.className = "rembutton";
-
         removeButton.onclick = () => {
             container.parentNode.removeChild(container);
             this.cardService.removeCard(this);
         };
 
         container.appendChild(displayTopic);
         container.appendChild(displayBody);
+        container.appendChild(this._dropDown);
+        container.appendChild(addTagButton);
+        container.appendChild(this._tagSection);
         container.appendChild(removeButton);
 
         this._dom = container;
     }
+
+    public resetDom(){
+    }
+
+    /**  
+     * By calling this, it will associate a specific tag with
+     * this object.
+     */
+    private addTag() {
+        var x: Tag = this.tagService.valueSelect(this._dropDown.options[this._dropDown.selectedIndex].value);
+        if (this.tags.indexOf(x) < 0){
+            this.tags.push(x);
+            this.publishTag(x);
+        }
+    }
+
+    private setTag(t: string) {
+        var x: Tag = this.tagService.valueSelect(t);
+        if (this.tags.indexOf(x) < 0){
+            this.tags.push(x);
+            this.publishTag(x);
+        }
+    }
+
+    /** 
+     * This will update the tags associated with the cards and create
+     * the <span> dom under the <div class="tag-Section">.
+     * @param t 
+     */
+    private publishTag(t: Tag){
+        const tag_item = document.createElement('span');
+        tag_item.className = 'tag-item';
+        tag_item.onclick = () => this.removeTag(t, tag_item);
+        tag_item.textContent = t.name;
+        console.log(t);
+        this._tagSection.appendChild(tag_item);
+        this.tagService.addAgent(() => {
+            if (!this.findTag(t)){
+                this.removeTag(t, tag_item);
+            }   
+        });
+    }
+
+    /**
+     * Removes given tag and dom object from it's parent node.
+     * @param t 
+     * @param e 
+     */
+    private removeTag(t: Tag, e: HTMLElement){
+        const index = this.tags.indexOf(t);
+        if (index > -1){
+            this.tags.splice(index, 1);
+            e.parentNode.removeChild(e);
+        }
+        else
+            throw ("Tag Not Found.");
```

Comment:
There is a huge bug in the program:
1. Create a new tag in the tag section.
2. Create a new card and add the newly created tag to it.
3. Now remove the tag which you created in step 1 from the tag section.
4. Try to create a new card.

You will not be able to create a new card and an exception will be thrown at you:
uncaught exception: Tag Not Found.